### ISSUE TITLE
Error starting notebook when environment variable BROWSER='firefox' is set

### ISSUE THREAD

**Issue by gabrielmagno** [jupyter/notebook#3837]
I tried starting a notebook, by running `jupyter notebook`. I was getting this error:

```
Traceback (most recent call last):
  File "/usr/bin/jupyter-notebook", line 11, in <module>
    load_entry_point('notebook==5.3.0', 'console_scripts', 'jupyter-notebook')()
  File "/usr/lib/python3.7/site-packages/jupyter_core/application.py", line 266, in launch_instance
    return super(JupyterApp, cls).launch_instance(argv=argv, **kwargs)
  File "/usr/lib/python3.7/site-packages/traitlets/config/application.py", line 658, in launch_instance
    app.start()
  File "/usr/lib/python3.7/site-packages/notebook/notebookapp.py", line 1597, in start
    browser = webbrowser.get(self.browser or None)
  File "/usr/lib/python3.7/webbrowser.py", line 42, in 
    register_standard_browsers()
  File "/usr/lib/python3.7/webbrowser.py", line 567, in register_standard_browsers
    cmd = _synthesize(cmdline, -1)
  File "/usr/lib/python3.7/webbrowser.py", line 116, in _synthesize
    register(browser, None, controller, update_tryorder)
TypeError: register() takes from 2 to 3 positional arguments but 4 were given

```
By looking at the file `webbrowser.py` I noticed that it could be related to the environment variable **BROWSER**. Mine is set to `BROWSER='firefox'`, then I tried other values to see if I could start jupyter notebook.

- `BROWSER=''`: works flawlessly, by opening in firefox, which is my default browser set in plasma desktop environment.
- `BROWSER='firefox'`: crashs, as mentioned before
- `BROWSER='konqueror'`: crashs, same error as in Firefox
- `BROWSER='google-chrome-stable'`: works flawlessly, by opening in Google Chrome
- `BROWSER='any_nonexistent_executable'`: works, but does not open any browser automatically

My setup:
- OS: Arch Linux
- Python: 3.7.0
- Jupyter: 4.1.0
- Jupyter Notebook: 5.3.0

**Comment by AlexisPrel** [comment]
I have the exact same configuration as you two @gabrielmagno and @alan1world  However starting the notebook with BROWSER set to "", although it launches the app, does not solve the issue: the kernel dies right away, with the following error message

```
Kernel started: 0df009b3-d364-4585-b680-f03bd1759180
Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/lib/python3.7/site-packages/ipykernel_launcher.py", line 16, in <module>
    app.launch_new_instance()
  File "/usr/lib/python3.7/site-packages/traitlets/config/application.py", line 658, in launch_instance
    app.start()
  File "/usr/lib/python3.7/site-packages/ipykernel/kernelapp.py", line 477, in start
    ioloop.IOLoop.instance().start()
  File "/usr/lib/python3.7/site-packages/tornado/platform/asyncio.py", line 132, in start
    self.asyncio_loop.run_forever()
  File "/usr/lib/python3.7/asyncio/base_events.py", line 510, in run_forever
    raise RuntimeError('This event loop is already running')
RuntimeError: This event loop is already running
```
I tried to find an issue on https://bugs.python.org/ that matches this but I am not familiar enough with the matter.

**Comment by alan1world** [comment]
@AlexisPrel  See [#3841](https://github.com/jupyter/notebook/issues/3841): This is an issue with an update to Tornado which occurred at the same time. If you do an update (sudo pacman -Syu) you'll get an update to python-ipykernel which will fix the tornado/asyncio error. Won't help with the BROWSER variable though!

**Comment by AlexisPrel** [comment]
Thank you so much, this works now!

**Comment by takluyver** [comment]
Closing as this should be fixed by Python 3.7.1 now. I don't think there's any practical way we can work around it in notebook, although affected users can unset the BROWSER environment variable.
